vital/set.h

```diff
@@ -94,6 +94,9 @@ class set
    */
   virtual T at( size_t index ) = 0;
   virtual T const at( size_t index ) const = 0;
+
+  virtual T operator[]( int index ) { return this->at( index ); }
```

Comment:
Traditionally, `operator[]` does not do bounds checking (and does not throw), while `at` does. Are we sure we want to not follow those semantics?

Reply:
What is the rationale for not bounds checking? If the index is out of bounds, the result is undefined. That sounds bad to me. I'm o.k. with [] checking and throwing to eliminate undefined behaviour.

Reply:
It adds an extra check that may not be needed (e.g. if the code has already done bounds check, or for other reasons knows that the index can't be out of range).